src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/ImplementGenericMathInterfacesCorrectlyTests.cs

```diff
@@ -0,0 +1,315 @@
+        {
+            await PopulateTestCs(@"
+using System;
+using System.Numerics;
+
+public class Test : IAdditionOperators<Test, MyTest, long>
+{
+    public static long operator +(Test left, MyTest right)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static long operator checked +(Test left, MyTest right)
+    {
+        throw new NotImplementedException();
+    }
+}
+public class MyTest : IAdditionOperators<[|Test|], MyTest, long> // 'IAdditionOperators' interface requires the derived type 'MyTest' used for the 'TSelf' type parameter
+{    public static long operator +(Test left, MyTest right)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static long operator checked +(Test left, MyTest right)
+    {
+        throw new NotImplementedException();
+    }
+}").RunAsync();
+        }
+
+        [Fact]
+        public async Task ParentClassImplementedIParsableShouldWarn()
+        {
+            await PopulateTestCs(@"
+using System;
+
+class Foo<TMe> : IParsable<TMe> where TMe : IParsable<TMe>
+{
+    public static TMe Parse(string s, IFormatProvider provider)
+    {
+        throw new NotImplementedException();
+    }
+    public static bool TryParse(string s, IFormatProvider provider, out TMe result)
+    {
+        throw new NotImplementedException();
+    }
+}
+class WrongImplementation : Foo<{|#0:int|}> { } // 'IParsable' interface requires the derived type 'WrongImplementation' used for the 'TMe' type parameter
+
+class CorrectImplementation : Foo<CorrectImplementation> { }
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMInterfacesRule).WithLocation(0).WithArguments("IParsable", "WrongImplementation", "TMe")).RunAsync();
+        }
+
+        [Fact]
+        public async Task ParentInterfaceImplementedIParsableShouldWarn()
+        {
+            await PopulateTestCs(@"
+using System;
+
+interface IMyInterface<TMe> : IParsable<TMe> where TMe : IParsable<TMe>
+{ }
+
+class WrongImplementation : IMyInterface<{|#0:int|}>
+{
+    public static int Parse(string s, IFormatProvider provider)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static bool TryParse(string s, IFormatProvider provider, out int result)
+    {
+        throw new NotImplementedException();
+    }
+}
+
+class CorrectImplementation : IMyInterface<CorrectImplementation>
+{
+    public static CorrectImplementation Parse(string s, IFormatProvider provider)
+    {
+        throw new NotImplementedException();
+    }
+
+    public static bool TryParse(string s, IFormatProvider provider, out CorrectImplementation result)
+    {
+        throw new NotImplementedException();
+    }
+}
+", VerifyCS.Diagnostic(ImplementGenericMathInterfacesCorrectly.GMInterfacesRule).WithLocation(0).WithArguments("IParsable", "WrongImplementation", "TMe")).RunAsync();
+        }
+
+        [Fact]
+        public async Task DerivedUsedBaseTypeaAsTypeParameters()
```

Comment:
I think there is a stray `a` in the name

`BaseTypeaAsType` => `BaseTypeAsType`